src/units/unit.hpp

```diff
@@ -1900,6 +1900,15 @@ class unit : public std::enable_shared_from_this<unit>
 	 * @param filter the filter used for checking.
 	 */
 	bool ability_matches_filter(const config & cfg, const std::string& tag_name, const config & filter) const;
+	/**
+	 * Check if an if filter_adjacent or filtr_adjacent_location matches
+	 * @param illuminates If ability if 'illuminates' type
+	 * @param cfg an ability WML structure
+	 * @param loc The location on which to resolve the ability
+	 * @param in_abilities_tag The tag_name of filter change idf simple abilities or specials or if abilities used like specials checking.
+	 * @returns true if it is active
+	 */
+	bool ability_active_adjacent_impl(bool illuminates, const config& cfg, const map_location& loc, bool in_abilities_tag = false) const;
```

Comment:
Does this need to be part of `unit`? It doesn't seem to rely on any class state, and either way should not be a public function.

Reply:
if because 'this' variable is used in filter_adjacent